nengo_loihi/inputs.py

```diff
@@ -14,17 +16,25 @@ class SpikeInput(LoihiInput):
     def __init__(self, n_neurons, label=None):
         super(SpikeInput, self).__init__(label=label)
         self.n_neurons = n_neurons
-        self.spikes = {}  # map sim timestep index to list of spike inds
+        self._ti = []
+        self._spikes = {}  # map sim timestep index to list of spike inds
 
     def add_spikes(self, ti, spike_idxs):
         assert is_integer(ti)
         ti = int(ti)
         assert ti > 0, "Spike times must be >= 1 (got %d)" % ti
-        assert ti not in self.spikes
-        self.spikes[ti] = spike_idxs
 
+        # Insert ti into self._ti in sorted order
+        ix = bisect_left(self._ti, ti)
+        assert (
+            ix == len(self._ti) or self._ti[ix] != ti
+        ), "spikes for time %d have already been added" % (ti,)
+        self._ti.insert(ix, ti)
+        self._spikes[ti] = spike_idxs
+
+    @property
     def spike_times(self):
-        return sorted(self.spikes)
+        return self._ti
```

Comment:
So the idea here is to keep the list of spike times sorted as we go, rather than sorting when requested? Why is this better? If we were requesting spike times a lot, I could see the rationale, but we only do it once per build (in the `build_input` function).

Reply:
Hm... I think because this PR was about speeding things up I assumed that this part of the diff was also about speeding things up, but looking back this was touched for other reasons. I'll revert.

Reply:
Reverted in 16f303151d107de9d8cdf3980160cd8bd7969a64.